# Exercise 02: Disparate Impact in Hiring Outcomes
# Cluster 3 - Statistics and Probability
# Framework: EU AI Act Annex III, NIST AI RMF (Manage)
# Case Study: Recruitment AI


In [3]:
import numpy as np

# Simulated hiring data: 100 applicants
np.random.seed(42)


women_scores=np.random.normal(loc=75,scale=8,size=50)
men_scores=np.random.normal(loc=75,scale=8,size=50)
print("Women scores (first 5):", women_scores[:5].round(2))
print("Men scores (first 5):", men_scores[:5].round(2))

Women scores (first 5): [78.97 73.89 80.18 87.18 73.13]
Men scores (first 5): [77.59 71.92 69.58 79.89 83.25]


In [9]:
selected_w = np.sum(women_scores > 75)
selected_m = np.sum(men_scores > 75)

hiring_rate_w = np.mean(women_scores > 75) * 100
hiring_rate_m = np.mean(men_scores > 75) * 100

print(f"Women hired: {selected_w}/50 ({hiring_rate_w:.1f}%)")
print(f"Men hired: {selected_m}/50 ({hiring_rate_m:.1f}%)")

Women hired: 20/50 (40.0%)
Men hired: 26/50 (52.0%)


In [13]:
Disparete_impact_ratio= hiring_rate_w /hiring_rate_m *100
if Disparete_impact_ratio>80:
   print(f"{Disparete_impact_ratio:.1f}% is not potential disparate impact.")
else:
    print(f"{Disparete_impact_ratio:.1f}% is potential disparate impact.")

76.9% is potential disparate impact.


**Exercise 02: Disparate Impact in Hiring Outcomes - Governance Reflection**

Disparate impact occurs when a policy or system appears fair on the surface but produces systematically unequal outcomes for different groups. In hiring AI, a system might score women and men identically on average, yet hire fewer women due to how the cutoff threshold interacts with sample variation. This hidden inequality is harder to detect than obvious bias  which is why it's a governance problem.

Fair scoring does not guarantee fair hiring outcomes because outcome fairness depends on multiple factors: the distribution of scores within each group, the cutoff threshold applied, and whether the system produces consistent results across different samples. In our simulation, women and men had identical average scores (75), yet only 40% of women were hired versus 52% of men. The difference emerged not from biased scoring, but from random variation in how many applicants from each group happened to score above the cutoff.

The 80% rule provides a legal threshold: if the hiring rate of one group is less than 80% of the hiring rate of another group, the system triggers potential disparate impact. Our result (76.9%) falls below this threshold, signaling a governance concern. Under EU AI Act Annex III High-Risk classification, disparate impact below 80% requires investigation. The company must answer: Is this outcome justified by genuine differences in applicant qualifications, or does it reflect system bias? If justified, they must disclose the outcome. If not justified, they must redesign the system or discontinue its use. This aligns with NIST AI RMF Manage function — auditing not just scoring fairness, but real-world hiring equity and accountability.

Under EU AI Act Annex III (High-Risk employment systems), the company must:

- Investigate: Why is the disparate impact happening? Is it the cutoff threshold? Is it the scoring weights? Is it something else?
- Document: Show what you found and what you're doing about it.
- Fix it: Either adjust the threshold, change the weights, or add human review to catch unfair outcomes.
- Test it: Prove the disparate impact ratio improves to 80% or above.
- Report it: If you can't fix it, you must disclose the limitation to users and regulators.

